ibm/resource_ibm_container_nlb_dns.go

```diff
@@ -0,0 +1,253 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/IBM-Cloud/container-services-go-sdk/kubernetesserviceapiv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIbmContainerNlbDns() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmContainerNlbDnsCreate,
+		ReadContext:   resourceIbmContainerNlbDnsRead,
+		UpdateContext: resourceIbmContainerNlbDnsUpdate,
+		DeleteContext: resourceIbmContainerNlbDnsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"cluster_name_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name or ID of the cluster. To list the clusters that you have access to, use the `GET /v1/clusters` API or run `ibmcloud ks cluster ls`.",
+			},
+			"nlb_host": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"nlb_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"nlb_dns_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"nlb_monitor_state": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"nlb_ssl_secret_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"nlb_ssl_secret_status": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"nlb_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"secret_namespace": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"x_auth_resource_group": &schema.Schema{
```

Comment:
rename the attribute to resource_group_id

Reply:
renamed to resource_group_id